python/cudf/cudf/core/tools/datetimes.py

```diff
@@ -479,7 +479,10 @@ def __init__(self, n=1, normalize=False, **kwds):
                 f"Keyword arguments '{','.join(list(wrong_kwargs))}'"
                 " are not yet supported in cuDF DateOffsets"
             )
-        self._scalars = _DateOffsetScalars(scalars)
+        # pandas DateOffset objects are made immutable by disabling
+        # `__setattr__`, but this class adds the _scalars attribute (also
+        # immutable after construction), so we bypass `DateOffset.__setattr__`.
+        object.__setattr__(self, "_scalars", _DateOffsetScalars(scalars))
```

Comment:
FWIW, we _don't_ disable `setattr` for `DateOffset` - so do we need this?

Reply:
`cudf.DateOffset` inherits from `pandas.DateOffset`, which does disable `__setattr__`. Previously, this was being overcome by defining `__setattr__` in this class to only allow setting of `_DateOffsetScalars` (as values, not as keys), which wasn't a very well targeted solution (you could do `obj.any_attribute_name = _DateOffsetScalars(...)` and it would work). Now I just defer to the parent implementation of disabling `__setattr__` and bypass it here.

Reply:
Ah right. Sorry, I thought our PR here was merged, where we remove the inheritance: https://github.com/rapidsai/cudf/pull/7078

Reply:
Got it, was not aware of that PR. Depending on which PR gets merged first, we can remove this block and just do `self._scalars = _DateOffsetScalars(scalars)`. I'll make a note on that PR too.